compiler/rustc_lint/src/builtin.rs

```diff
@@ -748,10 +750,40 @@ impl<'tcx> LateLintPass<'tcx> for MissingCopyImplementations {
         if def.has_dtor(cx.tcx) {
             return;
         }
+
+        // If the type contains a raw pointer, it may represent something like a handle,
+        // and recommending Copy might be a bad idea.
+        for field in def.all_fields() {
+            let did = field.did;
+            if cx.tcx.type_of(did).is_unsafe_ptr() {
+                return;
+            }
+        }
```

Comment:
While I think the rationale makes sense, it seems very inconsistent to me for pointers to actually be `Copy` but then we don't treat them as `Copy` in this lint. I would prefer that we don't make this change without a larger consensus that this is the right approach to take.

Reply:
Pointers themselves are just locations in memory with no semantics attached. However, the moment a user puts a pointer into their own type, they can add additional semantics, e.g., that the type uniquely borrows the data behind the pointer. For that reason, the decision to make a type with a pointer `Copy` is probably best made on a case-by-case basis, IMHO.

Reply:
We shouldn't recommend implementing `Copy` if might have certain other semantics. I think it's a reasonable assumption that when raw pointers are involved that is the case.

Of course that can never be prevented, for example some kind of  `Token` ZST not being `Copy` could be very intentional, but that is impossible to determine.

I'm happy to remove it if you disagree.